src/appmake/appmake.c

```diff
@@ -202,7 +202,7 @@ long parameter_search(const char *filen,const  char *ext,const char *target)
      /* Successfully opened the file so search through it.. */
     while ( fgets(buffer,LINEMAX,fp) != NULL ) {
         if (strncmp(buffer,target,strlen(target)) == 0 && isspace(buffer[strlen(target)])) {
-            sscanf(buffer,"%*s%*s%*[ $]%lx",&val);
+            sscanf(buffer,"%*s%*s%*[ $]%lx", (long unsigned int *) &val );
```

Comment:
I have a feeling this is a hangover from dos/32 bit - I think a better fix is probably `"%" PRIx32 ""`

Reply:
I'm not sure what you mean because all definitions of PRIx32 that I have found in C implementations are x or lx. Since both x and lx can only scan unsigned numbers, there must be a typecast between &var (being pointer to signed) and the result of the scan (being pointer to unsigned).

Since Z80 works with 16-bit addresses, all 16 bits of a scan with x or lx will be filled with the scanned number. So, if we want to represent an error as -1 in the same var, we need at least 32 bits, or a long type.

So, I would like to propose to stick with a temporary type casting to  (long unsigned int *) since this is cross C version and cross compiler compatible to my knowledge.

Grtz,
PB

Reply:
Sorry, that was my confusion, I thought it was a size issue rather than a signedness issue.